src/nvim/window.c

```diff
@@ -7614,19 +7613,50 @@ void win_get_tabwin(handle_T id, int *tabnr, int *winnr)
 
 void win_ui_flush(bool validate)
 {
+  kvec_t(win_T *) stack = KV_INITIAL_VALUE;
+
   FOR_ALL_TAB_WINDOWS(tp, wp) {
     if (wp->w_pos_changed && wp->w_grid_alloc.chars != NULL) {
       if (tp == curtab) {
+        // When a floating window is anchored to another window,
+        // update the position of its anchored window first.
+        while (true) {
+          wp->w_pos_changed = false;  // avoid looping forever in case of cycle
```

Comment:
should we WLOG or ILOG here?

Reply:
Now I've changed the implementation to use recursion instead, so adding a warning for loop does add complexity.